doc/source/index.rst

```diff
@@ -0,0 +1,333 @@
+
+command
+    Command used for concatenation.
+
+stdout
+    Optional; if command writes to stdout, redirect output here.
+
+publish_topic
+    Optional; publish a message when file is produced, using this topic.
+
+publish_port
+    Optional; use a custom port when publishing a message.
+
+nameservers
+    Optional; nameservers to publish on.
+
+subscriber_nameserver
+    Optional; nameserver to listen to.
+
+.. _Kai: https://navigator.eumetsat.int/product/EO:EUM:SW:METOP:165
+
+Example configuration:
+
+.. literalinclude:: ../../examples/cat.cfg_template
+   :language: ini
+
+catter
+^^^^^^
+
+Alternative to cat that does something else.
+
+.. literalinclude:: ../../examples/catter.cfg_template
+   :language: ini
+
+.. _gatherer:
+
+gatherer
+^^^^^^^^
+
+Watches files or messages and gathers satellite granules in "collections",
+sending then the collection of files in a message for further processing.
+Determines what granules with different start times belong together.
+A use case may be a reception system in which a single overpass results in
+multiple files, that should be grouped together for further processing.
+It uses `pytroll-schedule`_ to estimate the area coverage based on start
+and end times contained in filenames.
+
+The  ``gatherer`` collection is started when it receives a posttroll message,
+perhaps from `trollstalker`_ or `segment-gatherer`_.  Using the configured
+granule duration and the area of interest, it calculates the starting times
+of granules it should expect to be covered in this area before and after the
+granule it was messaged about.  Collection is considered finished when either
+all expected granules have been collected or when a timeout is reached,
+whatever comes first.  Timeout is configured with the ``timeliness`` option
+(see below).
+
+.. _pytroll-schedule: http://pytroll-schedule.readthedocs.org/
+
+The configuration file in INI format needs a section called ``[DEFAULT]``
+and one or more sections
+corresponding to what should be gathered.  The ``[DEFAULT]`` section should contain
+exactly one field:
+
+regions
+    A whitespace separated list of names corresponding to areas for
+    which granules are gathered.
+
+All other sections have the following mandatory fields:
+
+pattern
+    Defines the file pattern.  This is needed to create the full list of
+    expected files to know what to wait for.  If you don't pass this,
+    gatherer will not fail, but ...
+
+topics
+    Defines what posttroll topics to listen to for messages related to files
+    having arrived.
+
+publish_topic
+    Defines what posttroll topic shall be used to publish the news of all the
+    files that have been gathered.
+
+timeliness
+    Defines the maximum allowed age of the granule in minutes (Warning:
+    unit different compared to duration).  Collection is stopped
+    ``timeliness`` minutes after the expected end time of the last expected
+    granule.
+
+service
+    ??
```

Comment:
The `posttroll` service name which publishing the messages. If given, only subscribe to messages from this service. This *should* be an optional field.

Reply:
I had a failure when I didn't include it, but I don't remember the specifics.

Reply:
Moved it to optional and added description.